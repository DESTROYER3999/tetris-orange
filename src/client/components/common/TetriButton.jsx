import React from "react";
import styled from "styled-components";
import Cell from "../Cell";
import { CRIMSON, PURE_ORANGE } from "../../../constants/colors";

const StyledButton = styled.button`
  background: none;
  padding: 0px;
  margin: 0px;
  border: 0px;
  position: relative;
  top: -1020px;
  opacity: 0;
  transition: top 1s ease-out, transform 1s ease-out, opacity 1s;
  &:enabled {
    top: 0px;
    transform: rotate(720deg);
    opacity: 1;
  }
`;

const StyledTetriminos = styled.div`
  margin: 0;
  height: 40px;
  width: 160px;
  display: flex;
`;

const TetriButton = ({ disabled }) => {
  return (
    <StyledButton disabled={disabled} type="submit">
      <StyledTetriminos>
        <Cell color={PURE_ORANGE} letter="P"></Cell>
        <Cell color={CRIMSON} letter="L"></Cell>
        <Cell color={PURE_ORANGE} letter="A"></Cell>
        <Cell color={CRIMSON} letter="Y"></Cell>
      </StyledTetriminos>
    </StyledButton>
  );
};
export default TetriButton;
