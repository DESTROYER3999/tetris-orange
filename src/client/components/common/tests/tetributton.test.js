import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TetriButton from "../TetriButton";

describe("<TetriButton />", () => {
  test("TetriButton rendering without crashing", () => {
    const wrapper = shallow(<TetriButton />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
