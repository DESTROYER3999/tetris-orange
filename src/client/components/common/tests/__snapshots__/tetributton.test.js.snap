// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TetriButton /> TetriButton rendering without crashing 1`] = `
<styled.button
  type="submit"
>
  <styled.div>
    <Connect(CellSub)
      color="#ff8b00"
      letter="P"
    />
    <Connect(CellSub)
      color="#dc143c"
      letter="L"
    />
    <Connect(CellSub)
      color="#ff8b00"
      letter="A"
    />
    <Connect(CellSub)
      color="#dc143c"
      letter="Y"
    />
  </styled.div>
</styled.button>
`;
